projects/plugins/jetpack/_inc/lib/core-api/wpcom-endpoints/class-wpcom-rest-api-v2-endpoint-admin-menu.php

```diff
@@ -433,14 +433,14 @@ private function parse_menu_item( $title ) {
 		if ( false !== strpos( $title, 'count-' ) ) {
 			preg_match( '/<span class=".+\s?count-(\d*).+\s?<\/span><\/span>/', $title, $matches );
 
-			$count = absint( $matches[1] );
+			$count = absint( $matches[1] ? $matches[1] : 0 );
```

Comment:
```suggestion
			$count = absint(  isset( $matches[1] ) ? $matches[1] : 0 );
```

Maybe I should have done this?

Reply:
If you follow @anomiex's suggestion to move the `preg_match()` into the `if` condition, then there's no need for `isset()`, as `$matches[1]` will always be there. Otherwise, the code just won't go into this condition.

However, the regex `(\d*)` means that the `$matches[1]` might be an empty string, so `$matches[1] ? $matches[1] : 0` (or something simpler like `(int) $matches[1]`) still makes sense.

In general, conditions like `$matches[0] ? 'something' : 'something else'` will throw a warning if `$matches` doesn't exist or doesn't have the `[0]` element, because PHP will try to access it. As well as any other attempts to access a non-existing variable or array element.
So you're right, it will need `isset()` unless you modify the `if` condition.

You could also replace `absint()` with regular `(int)`. There's no need to force it into an absolute value, as the number will always be absolute anyway (`(\d*)` won't match anything but the numbers). No idea why it was done that way, there might've been a reason, but it's not here anymore 🙂

Reply:
Same goes for all the other two `if`'s below.